File: (8 kyu) Basic subclasses - Adam and Eve/(8 kyu) Basic subclasses - Adam and Eve.js
class God {
  /**
   * @returns Human[]
   */
  static create() {
    const Adam = new Man();
    const Eve = new Woman();
    return [Adam, Eve];
  }
}

class Human {
  constructor() {}
}

class Man extends Human {
  constructor() {
    super();
  }
}

class Woman extends Human {
  constructor() {
    super();
  }
}

File: (8 kyu) Basic subclasses - Adam and Eve/(8 kyu) Basic subclasses - Adam and Eve.java
public class God {
    public static Human[] create() {
        Human Adam = new Man();
        Human Eve = new Woman();
        return new Human[] { Adam, Eve };
    }
}

class Human {
    public Human() {
    }
}

class Man extends Human {
    public Man() {
    }
}

class Woman extends Man {
    public Woman() {
    }
}
